src/olympia/users/tasks.py

```diff
@@ -65,3 +77,49 @@ def update_user_ratings_task(data, **kw):
     )
     for pk, rating in data:
         UserProfile.objects.filter(pk=pk).update(averagerating=round(float(rating), 2))
+
+
+BATCH_SIZE = 100
+
+
+@task(autoretry_for=(HTTPError, Timeout), max_retries=5, retry_backoff=True)
+def sync_blocked_emails(batch_size=100, **kw):
+    url = f"""
+        {settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc
+    """.strip()
+    headers = {
+        'authorization': f'Bearer {settings.SOCKET_LABS_TOKEN}',
+    }
+    response = requests.get(url, headers=headers)
+
+    # Raise exception if not 200 like response
+    response.raise_for_status()
+
+    # Convert bytes to a file-like object
+    file_like_object = io.BytesIO(response.content)
+    # Create a Django ContentFile from the file-like object
+    django_file = ContentFile(file_like_object.read())
+    # Get 10 character uinique ID for file
+    file_id = str(uuid.uuid4())[:10]
+
+    # save csv to disk
+    file_path = default_storage.save(f'tmp/suppressions-{file_id}.csv', django_file)
```

Comment:
Can you use `tempfile.NamedTemporaryFile` or `tempfile.mkdtemp`? (both with `dir=settings.TMP_PATH` so it's stored in the appropriate place).

Reply:
Could you help me with a code snippet of exactly what you mean? :)

Reply:
it's difficult to write a complete snippet because it changes how you've written the rest of the function - you're using django storage rather than tempfile right now. 

Something like `with tempfile.NamedTemporaryFile(dir=settings.TMP_PATH, delete=not settings.DEBUG) as file_obj` and then write to the file_obj from `file_like_object`.

Or can you just use `csv.reader` to read directly from `file_like_object`? 🤔

Reply:
Okay so I tried both approaches. I ended up with using tempfile and then opening the tempfile inside the block. The original code was performing against the 21mb production csv at 10seconds. Trying to read the file_like_object directly tripled that. using tempfile brought it back down.

I would assume that reading the file directly in memory is less memory performant than reading from an actual file where we can stream chunks via the csv reader, but I don't know for sure. Added the change to a tmp commit that I'll squash before merging. LMK what you think.